Type error with GHC 9.2

Build with GHC 9.2 leads to the following type error

```
Build profile: -w ghc-9.2.1 -O0
In order, the following will be built (use -v for more details):
 - servant-0.18.3 (lib) (first run)
 - servant-0.18.3 (test:spec) (first run)
Preprocessing library for servant-0.18.3..
Building library for servant-0.18.3..
[31 of 33] Compiling Servant.Links    ( src/Servant/Links.hs, /home/guillaume/srcs/servant/dist-newstyle/build/x86_64-linux/ghc-9.2.1/servant-0.18.3/noopt/build/Servant/Links.o, /home/guillaume/srcs/servant/dist-newstyle/build/x86_64-linux/ghc-9.2.1/servant-0.18.3/noopt/build/Servant/Links.dyn_o )

src/Servant/Links.hs:652:10: error:
    • Uninferrable type variable k0 in
      type family equation right-hand side: (TypeError ...)
    • In the type instance declaration for ‘MkLink’
      In the instance declaration for ‘HasLink ((arr :: a -> b) :> sub)’
    |
652 |     type MkLink (arr :> sub) _ = TypeError (PartialApplication HasLink arr)
    |          ^^^^^^
cabal: Failed to build servant-0.18.3 (which is required by test:spec from
servant-0.18.3).
```

Associated with the following code:

```haskell
instance TypeError (PartialApplication HasLink arr) => HasLink ((arr :: a -> b) :> sub)
  where
    type MkLink (arr :> sub) _ = TypeError (PartialApplication HasLink arr)
    toLink = error "unreachable"
```

I solved the problem by just removing the instance (That's just better error, isn't it?.)

It may or may not come from: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.2#kind-inference-for-data-family-instances-is-pickier

--- a/servant/src/Servant/Links.hs
+++ b/servant/src/Servant/Links.hs
@@ -649,7 +649,7 @@ simpleToLink _ toA _ = toLink toA (Proxy :: Proxy sub)
 -- Erroring instance for 'HasLink' when a combinator is not fully applied
 instance TypeError (PartialApplication HasLink arr) => HasLink ((arr :: a -> b) :> sub)
   where
-    type MkLink (arr :> sub) _ = TypeError (PartialApplication HasLink arr)
+    type MkLink (arr :> sub) _ = TypeError (PartialApplication (HasLink :: * -> Constraint) arr)
     toLink = error "unreachable"
 
 -- Erroring instances for 'HasLink' for unknown API combinators

